src/hstreamdb_producer.erl

```diff
@@ -207,24 +217,37 @@ do_flush(Stream, OrderingKey, Records, Channel, Callback, State = #state{channel
     end.
 
 do_append_flush({OrderingKey, Record}, State = #state{stream = Stream,
+                                                      request_retry_interval_seconds = RequestRetryIntervalSeconds,
+                                                      append_retry_max_times = AppendRetryMaxTimes,
                                                       channel_manager = ChannelM}) ->
     case hstreamdb_channel_mgr:lookup_channel(OrderingKey, ChannelM) of
         {ok, Channel} ->
-            Res = flush_request(Stream, [Record], Channel),
+            Res = flush_request(Stream, [Record], Channel, RequestRetryIntervalSeconds, AppendRetryMaxTimes),
             {Res, State};
         {ok, Channel, NCManager} ->
-            Res = flush_request(Stream, [Record], Channel),
+            Res = flush_request(Stream, [Record], Channel, RequestRetryIntervalSeconds, AppendRetryMaxTimes),
             {Res, State#state{channel_manager = NCManager}};
         {error, Error} ->
             {{error, Error}, State}
     end.
 
-flush_request(Stream, Records, Channel) ->
+flush_request(Stream, Records, Channel, RequestRetryIntervalSeconds, AppendRetryMaxTimes) ->
     Req = #{streamName => Stream, records => Records},
     Options = #{channel => Channel},
+    case append_with_retry(Req, Options, RequestRetryIntervalSeconds, AppendRetryMaxTimes) of
+        {ok, Resp, _MetaData} ->
+            {ok, Resp};
+        {error, R} ->
+            {error, R}
+    end.
+
+append_with_retry(Req, Options, RequestRetryIntervalSeconds, AppendRetryMaxTimes) ->
     case hstreamdb_client:append(Req, Options) of
         {ok, Resp, _MetaData} ->
             {ok, Resp};
+        {error, {unavailable, _BinaryGrpcMessage}} when AppendRetryMaxTimes > 1 ->
+            timer:sleep(RequestRetryIntervalSeconds * 1000),
```

Comment:
Don't use `timer:sleep(N)`. Thread will sleep and cannot accept other call. Such as append more records.
Try
```erlang
erlang:send_after(Timeout, self(), {retry, Data})
```

If use `append_flush`, should use
```erlang
erlang:send_after(Timeout, self(), {retry, Data, From})
```
and after retry, reply caller as
```erlang
gen_server:reply(From, Result)
```
and return handle_call by {noreply, NextState}.

Maybe change much code.If you do not have much time . Just let me fix it.

Reply:
Does this means that I should add a clause in handle_info first?